src/tests/utils.cairo

```diff
@@ -34,13 +25,17 @@ fn pop_log<T, +Drop<T>, +starknet::Event<T>>(address: ContractAddress) -> Option
 
 /// Asserts that `expected_keys` exactly matches the indexed keys from `event`.
 ///
+/// This function removes the first key, which is the event ID, from the event
+/// so that only the event struct keys are checked.
+///
 /// `expected_keys` must include all indexed event keys for `event` in the order
 /// that they're defined.
 fn assert_indexed_keys<T, +Drop<T>, +starknet::Event<T>>(event: T, expected_keys: Span<felt252>) {
     let mut keys = array![];
     let mut data = array![];
 
     event.append_keys_and_data(ref keys, ref data);
+    let _ = keys.pop_front();
```

Comment:
Instead of removing the key, we should pass it as part of the expected keys. The first key if the event is not flattened is the `selector!("enum_member_name")`. We will test that events are not flattened with that approach.